Analysis/LeptonID/src/LeptonIDProcessor.cc

```diff
@@ -0,0 +1,374 @@
+                // could not find suitable mc particle, skip
+                continue;
+            }
+
+            // reverse check if mc particle also had most of its contributions in our reco particle
+            int MCReco_cw = 0;
+            int MCReco_tw = 0;
+            auto reco_candidate = getRelated(mcp_candidate, *MCTruthRecoNavigator, MCReco_cw, MCReco_tw);
+            if (reco_candidate != p || MCReco_cw < _MCTruthRecoCweightCut || MCReco_tw < _MCTruthRecoTweightCut) {
+                // p got most of its track/cluster hits from mcp but mcp contributed stronger to other reco particles, skip
+                continue;
+            }
+
+            // now cast mcp_candidate to a real MCParticle to use it
+            auto mcp = dynamic_cast<MCParticle *>(mcp_candidate);
+            _truePDG = mcp->getPDG();
+
+            auto parent = getMCParent(mcp);
+            int parentPDG = 0;
+            if (parent != nullptr) {
+                parentPDG = parent->getPDG();
+            }
+            _parentPDG = parentPDG;
+        }
+
+        TVector3 P(p->getMomentum());
+        _vars["seenP"] = P.Mag();
+        _vars["seenLogP"] = std::log(P.Mag());
+        _vars["seenCosTheta"] = P.CosTheta();
+        _vars["seenE"] = p->getEnergy();
+
+        auto dEdxParameters = pidh.getParticleID(p, dEdxAlgId).getParameters();
+        if (dEdxParameters.size() > (size_t) dEdxDist_e_Id) {
+            _vars["dEdxDist_e"] = dEdxParameters[dEdxDist_e_Id];
+        }
+
+        auto cluster = clusters[0];
+        getShowerShapes(cluster);
+
+        // TODO: access subdetector energies by name as for the shapes
+        auto subEnergies = cluster->getSubdetectorEnergies();
+        if (subEnergies.size() >= 2) {
+            _vars["seenEcalDep"] = subEnergies[0];
+            _vars["seenHcalDep"] = subEnergies[1];
+            if (subEnergies[0] > 0) {
+                _vars["ecal_share"] = subEnergies[0] / (subEnergies[0] + subEnergies[1]);
+            }
+        }
+        if (subEnergies.size() >= 3) {
+            _vars["seenYokeDep"] = subEnergies[2];
+        }
+
+        _vars["e_over_p"] = _vars["seenEcalDep"] / _vars["seenP"];
+
+        auto wgtp = getWeightedPoints3D(cluster, PandoraClusters);
+        getClusterShapes(wgtp);
+
+        if (_buildTree && std::isfinite(_vars["cluEllipsoid_r3"])) {
+            _tree->Fill();
+        }
+        if (_evalMVA) {
+            auto mvares = _mvareader.EvaluateMulticlass(_mvaname);
+            int pidClassID = std::distance(mvares.begin(), std::max_element(mvares.begin(), mvares.end()));
+
+            int pdg[3] = {11, 13, 211};
+
+            pidh.setParticleID(p, pidClassID, pdg[pidClassID], mvares[pidClassID], algoID, mvares);
+        }
+        clear();
+    }
+}
+
+void LeptonIDProcessor::check(LCEvent *evt)
+{
+    // nothing to check here - could be used to fill checkplots in reconstruction processor
+    (void) evt;
+}
```

Comment:
```suggestion
void LeptonIDProcessor::check(LCEvent *)
{
    // nothing to check here - could be used to fill checkplots in reconstruction processor
}
```
Easier way to get rid of unused variable warnings. ;)

Reply:
didn't know that one yet, thanks!